### PR TITLE
Housekeeping

### PR DESCRIPTION
**Pull request by nikku**
### Proposed Changes

On top of https://github.com/bpmn-io/vs-code-bpmn-io/pull/190 this applies additional housekeeping, including migration to ESLint 9 (https://github.com/bpmn-io/internal-docs/issues/1042).

### Checklist

To ensure you provided everything we need to look at your PR:

* [x] **Brief textual description** of the changes present
* [ ] **Visual demo** attached
* [ ] **Steps to try out** present, i.e. [using the `@bpmn-io/sr` tool](https://github.com/bpmn-io/sr)
* [x] Related issue linked via `Closes {LINK_TO_ISSUE}` or `Related to {LINK_TO_ISSUE}`

### CODE REVIEW

File: eslint.config.mjs [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+    '*.mjs'
+  ],
+  client: [
+    'src/client/**/*.ts',
+    'src/client/**/*.js'
+  ],
+  test: [
+    'src/test/*/*.ts'
+  ],
+  node_test: [
```
Comment (barmac) [inline eslint.config.mjs:17]: Quite unexpected 🐍

**Comment by barmac** [comment]
✅ with https://github.com/bpmn-io/vs-code-bpmn-io/pull/192/files#r1843872842 to consider

File: eslint.config.mjs (hunk 1 above)
Comment (nikku) [reply eslint.config.mjs:17]: What is unexpected there? Tests simply contain parts that are node, and parts that are _client_.

Comment (nikku) [reply eslint.config.mjs:17]: (Happy to follow-up if there is stuff we can improve).

Comment (barmac) [reply eslint.config.mjs:17]: snake_case, sorry for unclear comment

Comment (nikku) [reply eslint.config.mjs:17]: Ok :snake: Now I got your emoji.

**Pull request merged**